docs/design-patterns.mdx

```diff
@@ -5,66 +5,172 @@ title: Cadence Design Patterns
+    init() {
+        // hard-coded storage path: BAD
+        self.account.save(<-create Test(), to: /storage/testStorage)
+    }
+}
 
-### Example Code
+// Instead, use a field
+// Good practice!
+pub contract NamedFields {
+    pub resource Test {}
 
-See:
+    // field storage path: GOOD
+    pub let TestStoragePath: StoragePath
+
+    init() {
+        // assign and access the field here and in transactions
+        self.TestStoragePath = /storage/testStorage
+        self.account.save(<-create Test(), to: self.TestStoragePath)
+    }
+}
 
-[LockedTokens.cdc](https://github.com/onflow/flow-core-contracts/blob/bfb115869bd9f815cde1fe64ab6d91ca95c0938e/contracts/LockedTokens.cdc#L527)
+```
 
-[custody_setup_account_creator.cdc](https://github.com/onflow/flow-core-contracts/blob/79941fe65b634800065a440ae5243744b2ca8a2f/transactions/lockedTokens/admin/custody_setup_account_creator.cdc)
+[Example Code](https://github.com/onflow/flow-core-contracts/blob/master/contracts/LockedTokens.cdc#L583)
 
-[admin_deposit_account_creator.cdc](https://github.com/onflow/flow-core-contracts/blob/79941fe65b634800065a440ae5243744b2ca8a2f/transactions/lockedTokens/admin/admin_deposit_account_creator.cdc)
+## Script-Accessible Public Field/Function
 
-## Capability Revocation
+Data availability is important in a blockchain environment.
+It is useful to publicize information about your smart contract and the assets it controls
+so other smart contracts and apps can easily query it.
 
 ### Problem
 
-A capability provided by one account to a second account must able to be revoked by the first account without the co-operation of the second.
+Your contract, resource or struct has a field or resource that will need to be read and used on or off-chain, often in bulk.
 
 ### Solution
 
-The first account should create the capability as a link to a capability in `/private/`, which then links to a resource in `/storage/`. That first link is then handed to the second account as the capability for them to use. This can be stored in their private storage or a Capability Receiver.
+Make sure that the field can be accessed from a script (using a `PublicAccount`)
+rather than requiring a transaction (using an `AuthAccount`).
+This saves the time and soon the expense required by having to read a property using a transaction.
+Making the field or function `pub` and exposing it via a `/public/` capability will allow this.
 
-**Account 1:** `/private/capability` → `/storage/resource`
+Be careful not to expose any data or functionality that should be kept private when doing so.
+
+Example:
+
+```cadence
+// Field is private, so it cannot be read by the public
+// Bad
+access(self) let totalSupply: UFix64
+
+// Field is public, so it can be read and used by anyone
+// Good
+pub let totalSupply: UFix64
```

Comment:
```suggestion
// BAD: Field is private, so it cannot be read by the public
access(self) let totalSupply: UFix64

// GOOD: Field is public, so it can be read and used by anyone
pub let totalSupply: UFix64
```